src/AutoRest.CSharp/Mgmt/Decorator/ReferenceTypePropertyChooser.cs

```diff
@@ -26,6 +26,12 @@ internal static class ReferenceTypePropertyChooser
         private static readonly Type _resourceIdentifierType = typeof(Azure.Core.ResourceIdentifier);
         private static readonly Type _resourceTypeType = typeof(Azure.Core.ResourceType);
 
+        private static readonly IReadOnlyDictionary<string, string> _replacements = new Dictionary<string, string>
+        {
+            { "Azure.ResourceManager.CommonTypes.UserAssignedIdentity", "Azure.ResourceManager.Models.UserAssignedIdentity" },
+            { "Azure.ResourceManager.CommonTypes.ManagedServiceIdentity", "Azure.ResourceManager.Models.ManagedServiceIdentity" },
+        };
```

Comment:
I think we should put this in `TypeFactory` and let it returns an instance of `SystemObjectType` from this identity when it finds the crossLanguageDefinitionId of these
We should not let the typespec input part use these "replace types by shape" logic.

Reply:
updated